core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -386,6 +384,12 @@ public void onReceive(Context context, Intent intent) {
                 Downloader d = getDownloader(url);
                 if (d != null) {
                     d.cancel();
+                    DownloadRequester.getInstance().removeDownload(d.getDownloadRequest());
+
+                    FeedItem item = getFeedItemFromId(d.getDownloadRequest().getFeedfileId());
+                    if (item != null) {
+                        EventBus.getDefault().post(FeedItemEvent.updated(item));
+                    }
```

Comment:
It looks like here a canceled download is signaled by `FeedItemEvent` rather than `DownloadEvent`. In the same vein, I'd think #3045 should be redone as well. See https://github.com/AntennaPod/AntennaPod/pull/3045#issuecomment-536230004